### DISCUSSION TITLE
How to get the direction of turtle

### DISCUSSION THREAD

**Discussion by CaryCatZ** [cc-tweaked/CC-Tweaked#1284] [Q&A] [answered]
I want to make a automatic machine but there is a problem: I need to get the direction of turtle to resume from wrong exit without moving.

**Comment by Wojbie** [comment, marked as answer]
Simplest way to get direction of turtle is to use `gps` api to get turtle position. Move 1 block forward and use `gps` api again to get its new position.

Difference between those 2 positions will indicate in what direction is turtle facing. For example if x grows from 3 to 4 then turtle is facing positive x direction.
